include/swift/AST/DiagnosticsSema.def

```diff
@@ -7542,6 +7542,12 @@ NOTE(missing_several_cases,none,
 NOTE(missing_unknown_case,none,
      "handle unknown values using \"@unknown default\"", ())
 
+GROUPED_WARNING(decompose_default_case,StrictExhaustiveSwitches,none,
+  "switch can be made exhaustive without use of 'default'; replace with known cases", ())
+
+GROUPED_WARNING(default_after_catchall,StrictExhaustiveSwitches,none,
+                "a `default` case is equivalent to a catch-all case, remove one", ())
```

Comment:
Nit: diagnostics text uses single quotes rather than backticks; also, splice together separate clauses with semicolons.

Bigger picture: is a distinct warning necessary? seems like (unless `fallthrough` is used, but that's an exceedingly unlikely and more or less deliberate scenario) this would be covered by diagnosing that whatever comes second (of `case _` or `default`) is unreachable code.

Edit: there is also `redundant_particular_case` below, which seems to apply to this scenario too.